### ISSUE TITLE
Bug - mapping to existing destination - produces the same nested items inside collection

### ISSUE THREAD

**Issue by anisimovyuriy** [expressmapper/ExpressMapper#58]
Bug - mapping to existing destination - produces the same nested items inside collection:
Class declarations:
```//Source:
public class Source
{
    public SourceArrayContainer ArrayContainer {get;set;}
}
public class SourceArrayContainer
{
    public SourceNestedClass[] nestedArray1 {get;set;}
}
public class SourceNestedClass
{
    public string SourceNestedString {get;set;}
    public int SourceNestedInt {get;set;}
    public string sameNameNestedString {get;set;}
}
//Target:
public class Target
{
    public TargetArrayContainer ArrayContainer {get;set;}
}
public class TargetArrayContainer
{
    public TargetNestedClass[] NestedArray1 {get;set;}
}
public class TargetNestedClass
{
    public string TargetNestedString {get;set;}
    public int TargetNestedInt {get;set;}
    public string SameNameNestedString {get;set;}
}
```
Registration and mapping:
```Mapper.Reset();
Mapper.Register<SourceNestedClass, TargetNestedClass>()
    .Member(p => p.TargetNestedInt, p => p.SourceNestedInt)
    .Member(p => p.TargetNestedString, p => p.SourceNestedString);
Mapper.Register<SourceArrayContainer, TargetArrayContainer>();  
//ProcessAutoProperties() should pickup the array field
Mapper.Register<Source, Target>();

var source = new Source()
{
    ArrayContainer = new SourceArrayContainer()
    {
        nestedArray1 = new SourceNestedClass[]
        {
            new SourceNestedClass() {SourceNestedInt = 1},
            new SourceNestedClass(){SourceNestedString = "x", sameNameNestedString = "y"}
        }
    }
};
var res = Mapper.Map(source, new Target());
```

**Comment by anisimovyuriy** [comment]
duplicate #56

**Comment by anisimovyuriy** [comment]
Fixed and will be included in 1.7 release
